postgresdb/postgres.js

```diff
@@ -0,0 +1,81 @@
+const Pool = require('pg').Pool;
+const products = require('./config.js');
+
+const pool = new Pool (products);
+
+// for testing only
+const getAll = (req, res) => {
+  pool.query ('SELECT * FROM products ORDER BY id ASC', (err, data) => {
+    if (err) {
+      throw err;
+    }
+    res.status(200).json(data.rows);
+  })
+}
+
+const get = (req, res) => {
+  const id = parseInt(req.params.id, (err, data) => {
+    pool.query('SELECT * FROM products WHERE id = $1', [id], (err, data) => {
+      if (err) {
+        throw err;
+      }
+      res.status(200).json(data.rows);
+    })
+  })
+}
+
+const newEntry = (req, res) => {
+  const { productId, name, gender, category, styleId } = req.body;
+
+  pool.query(
+    'INSERT INTO products ( productId, name, gender, category, styleId ) VALUES ($1, $2, $3, $4, $5)',
+  [ productId, name, gender, category, styleId ],
+  (err, data) => {
+    if (err) {
+      throw err;
+    }
+    res.status(201).send(`Product added with ID: ${data.insertId}`);
+  })
+}
+
+const update = (req, res) => {
+  const { productId, name, gender, category, styleId } = req.body;
+
+  pool.query('UPDATE products SET productId = $1, name = $2, gender = $3, category = $4, styleId = $5 WHERE id = $1',
+  [name, email, id],
+  (error, results) => {
+    if (error) {
+      throw error
+    }
+    response.status(200).send(`Product modified with ID: ${id}`)
+  })
+}
+
+const remove = (req, res) => {
+  const id = parseInt(req.params.id);
+
+  pool.query('DELETE FROM products WHERE id = $1', [id], (err, data) => {
+    if (err) {
+      throw err;
+    }
+    res.status(200).send(`Product deleted with ID: ${id}`);
+  })
+}
+
+const newTable = (req, res) => {
+  pool.query('CREATE TABLE products (productId int, name varchar, gender varchar, category varchar, styleId int)', (err, data) => {
+    if (err) {
+      throw err;
+    }
+    console.log('Table created successfully');
+  })
+}
```

Comment:
What's the use case here?

Reply:
I put this in to create the table directly, but ended up doing it through command line instead. I may end up using this later, so I think I'll leave it for now.

Reply:
Although it’s a great method, a very well written might I add; I recommend removing it as we should try to follow the YAGNI engineering rule: https://en.m.wikipedia.org/wiki/You_aren't_gonna_need_it